pySPFM/workflows/pySPFM.py

```diff
@@ -543,120 +576,146 @@ def pySPFM(
-            hrf_norm = hrf_obj.generate_hrf().hrf_norm
-            estimates_spike = np.dot(np.tril(np.ones(nscans)), estimates_block)
-            fitts = np.dot(hrf_norm, estimates_spike)
-
     # Save activity-inducing signal
     if n_te == 1:
-        output_name = f"{output_filename}_beta.nii.gz"
+        output_name = get_outname(output_filename, "beta", "nii.gz", is_bids)
+        out_bids_keywords.append("beta")
     elif n_te > 1:
-        output_name = f"{output_filename}_DR2.nii.gz"
+        output_name = get_outname(output_filename, "DR2", "nii.gz", is_bids)
+        out_bids_keywords.append("DR2")
     write_data(
         estimates_spike,
         os.path.join(out_dir, output_name),
         mask,
         data_header,
         command_str,
         is_atlas=is_atlas,
+        is_bids=is_bids,
     )
 
     # Save fitts
     if n_te == 1:
-        output_name = f"{output_filename}_fitts.nii.gz"
+        output_name = get_outname(output_filename, "fitted", "nii.gz", is_bids)
+        out_bids_keywords.append("fitted")
         write_data(
             fitts,
             os.path.join(out_dir, output_name),
             mask,
             data_header,
             command_str,
             is_atlas=is_atlas,
+            is_bids=is_bids,
         )
     elif n_te > 1:
         for te_idx in range(n_te):
-            te_data = fitts[te_idx * nscans : (te_idx + 1) * nscans, :]
-            output_name = f"{output_filename}_dr2HRF_E0{te_idx + 1}.nii.gz"
+            te_data = fitts[te_idx * n_scans : (te_idx + 1) * n_scans, :]
+            output_name = get_outname(
+                output_filename, f"echo-0{te_idx + 1}_dr2HRF", "nii.gz", is_bids
+            )
+            out_bids_keywords.append(f"echo-0{te_idx + 1}_dr2HRF")
             write_data(
                 te_data,
                 os.path.join(out_dir, output_name),
                 mask,
                 data_header,
                 command_str,
                 is_atlas=is_atlas,
+                is_bids=is_bids,
             )
 
     # Save noise estimate
     if n_te == 1:
-        output_name = f"{output_filename}_MAD.nii.gz"
-        y = data_masked[:nscans, :]
-        _, _, noise_estimate = select_lambda(hrf=hrf_norm, y=y)
+        output_name = get_outname(output_filename, "MAD", "nii.gz", is_bids)
+        out_bids_keywords.append("MAD")
+        y = data_masked[:n_scans, :]
+        _, _, noise_estimate = select_lambda(hrf=hrf, y=y)
         write_data(
             np.expand_dims(noise_estimate, axis=0),
             os.path.join(out_dir, output_name),
             mask,
             data_header,
             command_str,
             is_atlas=is_atlas,
+            is_bids=is_bids,
         )
     else:
         for te_idx in range(n_te):
-            output_name = f"{output_filename}_MAD_E0{te_idx + 1}.nii.gz"
+            output_name = get_outname(
+                output_filename, f"echo-0{te_idx + 1}_MAD", "nii.gz", is_bids
```

Comment:
```suggestion
                output_filename, f"echo-{te_idx + 1}_MAD", "nii.gz", is_bids
```
I'd lean against a leading zero, since most of the time folks won't have enough echoes to make the leading zeros worthwhile.

Reply:
Sounds good!